pkg/management/authorisation.go

```diff
@@ -0,0 +1,342 @@
+	var action = "CreateClientRole"
+	var targetRealm = realmName
+
+	if err := c.checkAuthorisationOnTargetRealm(ctx, action, targetRealm); err != nil {
+		return "", err
+	}
+
+	return c.next.CreateClientRole(ctx, realmName, clientID, role)
+}
+
+func (c *authorisationComponentMW) checkAuthorisationOnTargetUser(ctx context.Context, action, targetRealm, userID string) error {
+	var accessToken = ctx.Value("access_token").(string)
+
+	// Retrieve the group of the target user
+
+	var userRep kc.UserRepresentation
+	var err error
+	if userRep, err = c.keycloakClient.GetUser(accessToken, targetRealm, userID); err != nil {
+		return ForbiddenError{}
+	}
+
+	if userRep.Groups == nil {
+		// No groups assigned, nothing allowed
+		return ForbiddenError{}
+	}
+
+	for _, targetGroup := range *userRep.Groups {
+		if c.checkAuthorisationOnTargetGroup(ctx, action, targetRealm, targetGroup) == nil {
+			return nil
+		}
+	}
+
+	return ForbiddenError{}
+}
+
+func (c *authorisationComponentMW) checkAuthorisationOnTargetGroup(ctx context.Context, action, targetRealm, targetGroup string) error {
+	var currentRealm = ctx.Value("realm").(string)
+	var currentGroups = ctx.Value("groups").([]string)
+
+	for _, group := range currentGroups {
+		targetGroupAllowed, wildcard := c.authorisations[currentRealm][group][action]["*"]
+
+		if wildcard {
+			_, allGroupsAllowed := targetGroupAllowed["*"]
+			_, groupAllowed := targetGroupAllowed[targetGroup]
+
+			if allGroupsAllowed || groupAllowed {
+				return nil
+			}
+		}
+
+		targetGroupAllowed, nonMasterRealmAllowed := c.authorisations[currentRealm][group][action]["/"]
+
+		if targetRealm != "master" && nonMasterRealmAllowed {
+			_, allGroupsAllowed := targetGroupAllowed["*"]
+			_, groupAllowed := targetGroupAllowed[targetGroup]
+
+			if allGroupsAllowed || groupAllowed {
+				return nil
+			}
+		}
+
+		targetGroupAllowed, realmAllowed := c.authorisations[currentRealm][group][action][targetRealm]
+
+		if realmAllowed {
+			_, allGroupsAllowed := targetGroupAllowed["*"]
+			_, groupAllowed := targetGroupAllowed[targetGroup]
+
+			if allGroupsAllowed || groupAllowed {
+				return nil
+			}
+		}
+	}
+
+	return ForbiddenError{}
+}
+
+func (c *authorisationComponentMW) checkAuthorisationOnTargetRealm(ctx context.Context, action, targetRealm string) error {
+	var currentRealm = ctx.Value("realm").(string)
+	var currentGroups = ctx.Value("groups").([]string)
+
+	for _, group := range currentGroups {
+		_, wildcard := c.authorisations[currentRealm][group][action]["*"]
+		_, nonMasterRealmAllowed := c.authorisations[currentRealm][group][action]["/"]
+		_, realmAllowed := c.authorisations[currentRealm][group][action][targetRealm]
+
+		if wildcard || realmAllowed || (targetRealm != "master" && nonMasterRealmAllowed) {
+			return nil
+		}
+	}
+
+	return ForbiddenError{}
+}
+
+// ForbiddenError when an operation is not permitted.
```

Comment:
If we use something like this below sample code, we can avoid adding a case in managementErrorHandler() for each new error and avoid unit tests.
```
var ForbiddenError = HTTPError{
	Status:  http.StatusForbidden,
	Message: "ForbiddenError: Operation not permitted.",
}
```
or less 'pretty' but more generic, something like the CreateForbiddenError(...) like the existing CreateMissingParameterError(...)